tests/test_apply_variants.py

```diff
@@ -498,111 +630,188 @@ def test_insertion_at_interval_ends(self):
+
+        reference_alignment = apply_variants(genome37.dna, variants, positive_strand_interval, reference_alignment=True)[1]
+        self.assertEqual(reference_alignment, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+
+        reference_alignment = apply_variants(genome37.dna, variants, negative_strand_interval, reference_alignment=True)[1]
         self.assertEqual(reference_alignment, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
 
         # Test anchor < start with substitution
-        interval = Interval('chr1', '+', 5, 15, genome37, 0)
-        reference_alignment = apply_variants(genome37.dna, variants, interval, reference_alignment=True)[1]
+        positive_strand_interval = Interval('chr1', '+', 5, 15, genome37, 0)
+        negative_strand_interval = Interval('chr1', '-', 5, 15, genome37, 0)
+
+        reference_alignment = apply_variants(genome37.dna, variants, positive_strand_interval, reference_alignment=True)[1]
+        self.assertEqual(reference_alignment, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+
+        reference_alignment = apply_variants(genome37.dna, variants, negative_strand_interval, reference_alignment=True)[1]
         self.assertEqual(reference_alignment, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
 
         # Test anchor < start with substitution
         variants = [Variant.from_string("chr1:11::TT", self.genome)]
-        interval = Interval('chr1', '+', 5, 15, genome37, 0)
-        reference_alignment = apply_variants(genome37.dna, variants, interval, reference_alignment=True)[1]
+        positive_strand_interval = Interval('chr1', '+', 5, 15, genome37, 0)
+        negative_strand_interval = Interval('chr1', '-', 5, 15, genome37, 0)
+
+        reference_alignment = apply_variants(genome37.dna, variants, positive_strand_interval, reference_alignment=True)[1]
         self.assertEqual(reference_alignment, [0, 1, 2, 3, 4, (5, 0), (5, 1), 5, 6, 7])
 
+        reference_alignment = apply_variants(genome37.dna, variants, negative_strand_interval, reference_alignment=True)[1]
+        self.assertEqual(reference_alignment, [2, 3, 4, (5, 0), (5, 1), 5, 6, 7, 8, 9])
+
         # Test anchor right of deletion
         variants = [Variant.from_string("chr1:11:GT:", self.genome)]
-        interval = Interval('chr1', '+', 5, 15, genome37, 11)
-        reference_alignment = apply_variants(genome37.dna, variants, interval, reference_alignment=True)[1]
+
+        positive_strand_interval = Interval('chr1', '+', 5, 15, genome37, 11)
+        negative_strand_interval = Interval('chr1', '-', 5, 15, genome37, 11)
+
+        reference_alignment = apply_variants(genome37.dna, variants, positive_strand_interval, reference_alignment=True)[1]
         self.assertEqual(reference_alignment, [-1, 0, 1, 2, 3, 4, 6, 7, 8, 9])
 
+        reference_alignment = apply_variants(genome37.dna, variants, negative_strand_interval, reference_alignment=True)[1]
+        self.assertEqual(reference_alignment, [0, 1, 2, 3, 5, 6, 7, 8, 9, 10])
```

Comment:
I might be getting confused but I'm not sure the original case is correct even: I see a deletion of 2, where the anchor is in the middle?

```
5p-->3p +'ve

 5  6  7  8  9 10 11 12 13 14 15

 0  1  2  3  4  5  6  7  8  9 10
  C  G  T  A  C  G  T  A  C  G
  G  C  A  T  G  C  A  T  G  C
10  9  8  7  6  5  4  3  2  1  0

deleting pos [10,12) should delete 5,6 

 4  5  6  7  8  9 10 11 12 13 14 15 16
-1  0  1  2  3  4  5  6  7  8  9 10 11
  A  C  G  T  A  C  - |-  A  C  G  T
  
  T  G  C  A  T  G  - |-  T  G  C  A
11 10  9  8  7  6  5  4  3  2  1  0 -1
```

Reply:
i'll take a look later but likely it's not splitting the var over apply left + right,

Reply:
https://github.com/deepgenomics/GenomeKit/pull/140

Reply:
will need a merge/rebase from master now